lib/src/form_builder_validators.dart

```diff
@@ -389,19 +389,25 @@ class FormBuilderValidators {
   }
 
   /// [FormFieldValidator] that requires the field's value to be a valid email address.
-  /// This validator checks if the field's value is a valid email address.
-  ///
-  /// ## Parameters:
-  /// - [errorText] The error message to display when the email is invalid.
   ///
   /// {@macro email_template}
+  /// {@macro email_regex_template}
   static FormFieldValidator<String> email({
     String? errorText,
+    RegExp? emailRegex,
+    bool invalidNullAndEmpty = false,
   }) =>
-      (valueCandidate) =>
-          (valueCandidate?.isNotEmpty ?? false) && !isEmail(valueCandidate!)
-              ? errorText ?? FormBuilderLocalizations.current.emailErrorText
-              : null;
+      invalidNullAndEmpty
+          ? EmailValidator(
+              errorText:
+                  errorText ?? FormBuilderLocalizations.current.emailErrorText,
+              emailRegex: emailRegex,
+            ).call
+          : EmailValidator(
```

Comment:
If you look at the syntax like this, it is some duplication. It would be better to call the constructor of the validator with `validateNullOrEmpty` or something, but i'm not sure that is possible. In C# i would create method overloads or call super within the method.